### DISCUSSION TITLE
Use debugging with a project running in devcontainer and not using vscode?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [devcontainers/spec#231] [Q&A]
Would it be possible to spinnup a devcontainer and then debug code in it without using vscode?

For example I have a freq trade devcontainer and want to debug some python scripts in it that require the freqtrade trading bot to run. The strategy.py is loaded by the freqtrade bot and i would like to set breakpoints in this strategy.py that get triggered when the freqtrade bot is loading and running the strategy.

Is this something that is possible?

The reason for asking is that I would like rather use IntelliJ to debug and edit my files ( i work with it for 15 years ) and cannot get used to vscode UI.

**Comment by bamurtaugh** [comment]
Hi @Jarrodsz, thanks for opening this!

You could use GitHub Codespaces to help achieve this: https://docs.github.com/en/codespaces/developing-in-codespaces/using-github-codespaces-in-your-jetbrains-ide. Please let us know if this seems like it'd help or you have other questions! cc @joshspicer

**Comment by Jarrodsz** [comment]
Hi thank your for pointing this out diden't knew it existed even.

Scanning the docs it seems one needs to have the code in a remote repostitory? that is not a wanted situation. 
I want to connect to the local devcontainer without having to push the code to github, do i miss something or is that the only way to get it working?

**Reply by joshspicer** [reply]
Codespaces is a cloud hosted service, so at some point you'd need to either push your code to a GitHub repo (the normal/happy path), or otherwise have some kind of bootstrapping script to pull your code into the Codespace.  

I know folks in the past that have had their code hosted in Azure Devops or some other remote repository, and have used the `onCreateCommand` lifecycle hook to do the clone.  I'd definitely recommend keeping everything in GitHub if you chose to use codespaces, but if you wanted to get more creative there are options for you to explore :)

**Comment by Jarrodsz** [comment]
Thank you for the info.

But my usecase is just connecting intellij to
The devcontainer so you can debug just like in vscode. It would just connect to the local devcontainer instead of remote code locations.

I basically want to mimic the devcontainer as it runs in vscode.

Vscode also automatically forward ports ehen a service gets started in my case a trading bot on port 8080 and up.

Im using vscode on dev but the idea is to use a devcontainer on a remote vps thats why im trying to remove the vscode dependency you see.
